tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorTests.cs

```diff
@@ -120,6 +120,118 @@ public void Any_with_a_predicate_on_documents()
                 "{ G : { $elemMatch : { D : \"Don't\", 'E.F' : 33 } } }");
         }
 
+        [Fact]
+        public void Any_with_a_predicate_on_document_itself()
+        {
+            Assert(
+                x => x.G.Any(g => g != null),
+                2,
+                "{ 'G' : { '$elemMatch' : { '$ne' : null } } }");
+
+            Assert(
+                x => x.G.Any(g => null != g),
+                2,
+                "{ 'G' : { '$elemMatch' : { '$ne' : null } } }");
+
+            Assert(
+                x => x.G.Any(g => g.E.I.Any(i => i == "insecure")),
+                1,
+                "{ \"G.E.I\" : { '$elemMatch' : { '$eq': 'insecure' } } }");
```

Comment:
I'm suspicious of this flattening... how did TWO nested Anys turn into ONE $elemMatch?

Reply:
the same as this one:

            Assert(
                x => x.G.Any(g => g.S.Any(s => s.D == "Delilah")),
                1,
                "{ \"G.S\" : { $elemMatch : { \"D\" : \"Delilah\" } } }");

I can check if you want how it is generated, but it's not in scope of this ticket. 
The reason of why I've added this test, it's because nested `Any` predicate uses the input query parameter itself, without any `.` part.

Reply:
OK... if it is not in scope of this ticket we can ignore.

It's just that I would have expected that nested Any calls in LINQ would have resulted in nested $elemMatch in MongoDB, but I have no idea how this code works or what magic ended up flattening two nested Any into one single $elemMatch.